libclamav_rust/src/fuzzy_hash.rs

```diff
@@ -0,0 +1,544 @@
+// Sample c-facing function call
+
+use image::{imageops::FilterType::Lanczos3, DynamicImage, ImageBuffer, Luma, Pixel, Rgb};
+use num_traits::{NumCast, ToPrimitive, Zero};
+use rustdct::DctPlanner;
+use transpose::transpose;
+
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::ffi::CStr;
+use std::mem::ManuallyDrop;
+use std::os::raw::c_char;
+use std::slice;
+
+use log::*;
+use thiserror::Error;
+
+use crate::sys;
+use crate::validate_str_param;
+
+/// CdiffError enumerates all possible errors returned by this library.
+#[derive(Error, Debug)]
+pub enum FuzzyHashError {
+    #[error("Invalid format")]
+    Format,
+
+    #[error("Unknown algorithm: {0}")]
+    UnknownAlgorithm(String),
+
+    #[error("Invalid hamming distance: {0}")]
+    FormatHammingDistance(String),
```

Comment:
This should be the numeric type for hamming distance.  If the hamming distance is not a valid number, that would be a different error.

Reply:
`FormatHammingDistance(<some integer type>)` is what I meant there, if it wasn't clear.